### PR TITLE
fix: public applications and request content-type

### PR DESCRIPTION
**Pull request by lcpojr**
We forgot to add the content-type on tests and i've inverted the confidential validation.

### CODE REVIEW

File: apps/authenticator/lib/sign_in/commands/resource_owner.ex [hunk 1]
```diff
@@ -96,6 +91,11 @@ defmodule Authenticator.SignIn.Commands.ResourceOwner do
         FakeVerifyHash.execute(:argon2)
         {:error, :unauthenticated}
 
+      {:public?, false} ->
+        Logger.info("Client application #{client_id} is not public")
+        FakeVerifyHash.execute(:argon2)
+        {:error, :unauthenticated}
```
Comment (lcpojr) [inline apps/authenticator/lib/sign_in/commands/resource_owner.ex:97]: Here i'll request the `client_assertion` in another pull request.

Comment (caiaffa) [reply apps/authenticator/lib/sign_in/commands/resource_owner.ex:97]: Ok

Comment (caiaffa) [reply apps/authenticator/lib/sign_in/commands/resource_owner.ex:97]: Failed  code consistence: 
https://github.com/lcpojr/watcher_ex/pull/53/checks?check_run_id=2774855844

**Review by caiaffa — CHANGES_REQUESTED** [review]
Failed code consistence:
https://github.com/lcpojr/watcher_ex/pull/53/checks?check_run_id=2774855844

**Pull request merged**